core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java

```diff
@@ -541,12 +540,12 @@ public void testRowFilterOnHiddenColumn()
                 .matches("VALUES BIGINT '25'");
         // TODO https://github.com/trinodb/trino/issues/10006 - support insert into a table with row filter that is using hidden columns
         assertThatThrownBy(() -> assertions.query("INSERT INTO mock.tiny.nation_with_hidden_column VALUES (101, 'POLAND', 0, 'No comment')"))
-                .isInstanceOf(IllegalArgumentException.class)
-                .hasMessage("scope: 5, fields mappings: 4");
+                .isInstanceOf(ArrayIndexOutOfBoundsException.class)
+                .hasMessage("Index 4 out of bounds for length 4");
```

Comment:
Maybe we can improve the error message from the user perspective?

Reply:
It is way harder to fix. I would need to rewrite a lot of planer code for insert. Same for error message.